src/overlays/actors/ovl_En_Holl/z_en_holl.c

```diff
@@ -24,37 +36,265 @@ const ActorInit En_Holl_InitVars = {
+                                                               absRotatedPlayerZ > EN_HOLL_LOADING_PLANE_DISTANCE) {
+            s32 enHollId = EN_HOLL_GET_ID_CAST(this);
+            s32 playerSide = (rotatedPlayerPos.z < 0.0f) ? EN_HOLL_BEHIND : EN_HOLL_BEFORE;
+            TransitionActorEntry* transitionActorEntry = &globalCtx->doorCtx.transitionActorList[enHollId];
+            s8 room = transitionActorEntry->sides[playerSide].room;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899960.s")
+            this->actor.room = room;
+            if ((this->actor.room != globalCtx->roomCtx.currRoom.num) &&
+                (Room_StartRoomTransition(globalCtx, &globalCtx->roomCtx, this->actor.room))) {
+                this->actionFunc = EnHoll_RoomTransitionIdle;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_808999B0.s")
+void EnHoll_VerticalBgCoverIdle(EnHoll* this, GlobalContext* globalCtx) {
+    f32 absYDistToPlayer;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/EnHoll_Init.s")
+    if ((this->actor.xzDistToPlayer < EN_HOLL_RADIUS) &&
+        (absYDistToPlayer = fabsf(this->actor.yDistToPlayer),
+         absYDistToPlayer < EN_HOLL_ACTIVATION_PLANE_DISTANCE_VERTICAL)) {
+        if (absYDistToPlayer < EN_HOLL_LOADING_PLANE_DISTANCE_VERTICAL) {
+            globalCtx->bgCoverAlpha = 255;
+        } else {
+            globalCtx->bgCoverAlpha = EN_HOLL_SCALE_BG_COVER_ALPHA(absYDistToPlayer);
+        }
+        if (absYDistToPlayer > EN_HOLL_LOADING_PLANE_DISTANCE_VERTICAL) {
+            s32 enHollId = EN_HOLL_GET_ID_CAST(this);
+            s32 playerSide = (this->actor.yDistToPlayer > 0.0f) ? EN_HOLL_ABOVE : EN_HOLL_BELOW;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/EnHoll_Destroy.s")
+            this->actor.room = globalCtx->doorCtx.transitionActorList[enHollId].sides[playerSide].room;
+            if ((this->actor.room != globalCtx->roomCtx.currRoom.num) &&
+                Room_StartRoomTransition(globalCtx, &globalCtx->roomCtx, this->actor.room)) {
+                this->actionFunc = EnHoll_RoomTransitionIdle;
+                this->bgCoverAlphaActive = true;
+            }
+        }
+    } else if (this->bgCoverAlphaActive) {
+        this->bgCoverAlphaActive = false;
+        globalCtx->bgCoverAlpha = 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899ACC.s")
+void EnHoll_VerticalIdle(EnHoll* this, GlobalContext* globalCtx) {
+    f32 absYDistToPlayer;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899B88.s")
+    if ((this->actor.xzDistToPlayer < EN_HOLL_RADIUS) &&
+        (absYDistToPlayer = fabsf(this->actor.yDistToPlayer),
+         absYDistToPlayer < EN_HOLL_ACTIVATION_PLANE_DISTANCE_VERTICAL) &&
+        (absYDistToPlayer > EN_HOLL_LOADING_PLANE_DISTANCE_VERTICAL)) {
```

Comment:
Errr, this looks like a bit overcomplicated, can you try to split this if condition in 2?
Something like this:
```suggestion
    if (this->actor.xzDistToPlayer < EN_HOLL_RADIUS)) {
        absYDistToPlayer = fabsf(this->actor.yDistToPlayer);
        if (absYDistToPlayer < EN_HOLL_ACTIVATION_PLANE_DISTANCE_VERTICAL) &&
            (absYDistToPlayer > EN_HOLL_LOADING_PLANE_DISTANCE_VERTICAL)) {
```

Reply:
Pretty sure this doesn't match. The assignment actually has to be in the conditional to match, based on the comments in OoT's version of EnHoll.

Reply:
Wild. I guess it does match with MM's compiler flags.

Reply:
Is it worth diverging from OoT's EnHoll to rewrite it as two separate IFs like this? I kinda like it how it is, honestly. If only because that's the only thing that works in OoT (debug).

Reply:
Also I kind of like defining that variable in the IF scope, since it's never used outside the if.

Reply:
Okay, so while this suggested change does match in EnHoll_VerticalBgCoverIdle(), it does not match in EnHoll_TransparentIdle(), even in MM with retail compiler flags. Surely it's better to be consistent throughout the entire file? Or would we rather have one idle func slightly more readable at the cost of inconsistency?

Reply:
Hmmm, I think I prefer the readability over the inconsistency in this case. That other function may need other changes to remove that fake match

Reply:
I'm reasonably sure it's not a fake match. Using the comma in the IF is swaps the order of two instructions, which is necessary for a match in EnHoll_TransparentIdle() in both OoT and MM.

Reply:
I can def go ahead and remove the comma in both EnHoll_VerticalBgCoverIdle() and EnHoll_VerticalIdle() though, and leave it in for EnHoll_TransparentIdle() if that's what you prefer.

Reply:
I personally would prefer it that way, but I think this is up to you. So, whatever style you prefer.

Reply:
actually, nope, i can't get comma removal it to match in EnHoll_VerticalBgCoverIdle either.... The only one I've been able to match it in is EnHoll_VerticalIdle. I can go ahead and commit that one though if you'd like. But VerticalBgCoverIdle and TransparentIdle will still have the comma.

Reply:
I removed the only comma I could as of commit https://github.com/zeldaret/mm/pull/270/commits/6c376507e406c3eacd10a4b4ed74ea4ee83381bc so I'm going to mark this one as resolved.